src/Models_Functions/MTV/Compute_M0_T1_OnSPGR.m

```diff
@@ -0,0 +1,83 @@
+function [T1, M0] = Compute_M0_T1_OnSPGR(data, flipAngles, TR, b1Map, roi, verbose)
+%Perform a simple linear-least squares data fit on variable flip angle SPGR data 
+%
+% function [M0, T1] = Compute_M0_T1_OnSPGR(data, flipAngles, TR [, b1Map, roi, verbose])
+% -----------------------------------------------------------
+% INPUTS:
+%   data: width x length x slices x flipAngles matrix
+%   flipAngles: vector of flip angles (in degrees) corresponding to last dimension of 'data'
+%   TR: Repetition time in seconds
+%   b1Map: width x length x slices matrix containing relative flip angle
+%         (i.e. if nominal alpha is 60 and measured alpha is 61, then b1Map = 61/60
+%   roi: width x length x slices binary mask 
+%   verbose: logical - for debugging
+
+
+if ndims(data)<3, data = permute(data(:),[2 3 4 1]); end
+[nX, nY, nZ, nFlip] = size(data);
+nVox = nX*nY*nZ;
+
+if (nargin < 4) || isempty(b1Map)
+    b1Map = ones(nX, nY, nZ);
+end
+
+if nargin<5 || isempty(roi)
+    roi = true(nX, nY, nZ);
+end
+
+if nargin<6
```

Comment:
I've been switching out of using nargin to handle default function parameters, as it is so easy to forget the proper numbering when handling them and refacturing. For example, currently your function has 5 input arguments, but you are handling the case for 6. It appears you missed adding your mask (i.e. ROI) at some point.

I prefer this alternative:

```matlab
if ~exist('myParam', 'var')
    myParam = DEFAULT
end
```

That way, if ever you add/remove function arguments, you don't have to remember to check which number they are set at.

Reply:
What do you think?

Reply:
Resolved in fac50e0a

Reply:
That seems much better than using `nargin`. Clearer too. 
Will update in next commit.